src/DonationForms/ServiceProvider.php

```diff
@@ -80,6 +89,121 @@ public function boot()
             RemoveDuplicateMeta::class,
             UpdateDonationLevelsSchema::class,
         ]);
+
+        $this->registerAsyncData();
+    }
+
+    /**
+     * @unreleased
+     */
+    private function registerAsyncData()
+    {
+        // Async ajax request
+        Hooks::addAction('wp_ajax_givewp_get_form_async_data_for_list_view', GetAsyncFormDataForListView::class);
+        Hooks::addAction('wp_ajax_nopriv_givewp_get_form_async_data_for_list_view', GetAsyncFormDataForListView::class);
+
+        // Filter from give_goal_progress_stats() function which is used by the admin form list views and form grid view
+        Hooks::addFilter('give_goal_progress_stats', GiveGoalProgressStats::class,
+            'maybeChangeGoalProgressStatsActualValue', 999,
+            2);
+
+        // Only register assets on the frontend, but not enqueue to prevent loading them in unnecessary places
+        Hooks::addAction('wp_enqueue_scripts', LoadAsyncDataAssets::class, 'registerAssets');
+        add_action('give_before_template_part', function ($templateName) {
+            if ('shortcode-form-grid' === $templateName) {
+                // Enqueue assets previously registered on demand - only when the shortcode gets rendered
+                LoadAsyncDataAssets::enqueueAssets();
+            }
+        });
+
+        // Form Grid
+        add_filter('give_form_grid_goal_progress_stats_before', function () {
+            $usePlaceholder = give(FormGridView::class)->maybeUsePlaceholderOnGoalAmountRaised();
+
+            if ($usePlaceholder) {
+                //Enable placeholder on the give_goal_progress_stats() function
+                add_filter('give_goal_progress_stats', function ($stats) {
+                    $stats['actual'] = AsyncDataHelpers::getSkeletonPlaceholder('1rem');
+
+                    return $stats;
+                });
+                add_filter('give_goal_shortcode_stats', function ($stats) {
+                    $stats['income'] = 0;
+
+                    return $stats;
+                });
+            }
+        });
+        Hooks::addFilter('give_form_grid_progress_bar_amount_raised_value', FormGridView::class, 'maybeSetProgressBarAmountRaisedAsync',10,2);
+        Hooks::addFilter('give_form_grid_progress_bar_donations_count_value', FormGridView::class, 'maybeSetProgressBarDonationsCountAsync',10,2);
+
+        // Load assets on the admin list form pages
+        $isLegacyAdminFormListPage = isset($_GET['post_type']) && 'give_forms' === $_GET['post_type'] && !isset($_GET['page']);
+        $isAdminFormListPage = isset($_GET['post_type']) && 'give_forms' === $_GET['post_type'] && isset($_GET['page']) && 'give-forms' === $_GET['page'];
```

Comment:
@glaubersilva sorry this still does not account for the initial page when clicking the "Donations" menu item:
This page does not have the `post_type` param, just the `page` arg:

```
isset($_GET['page']) && $_GET['page'] === 'give-forms'
```

Ref: https://github.com/impress-org/givewp/blob/f83c6c9bc01a77a4ddc76b8adb0b3816e97d4e14/src/DonationForms/V2/DonationFormsAdminPage.php#L268

<img width="1509" alt="Screenshot 2024-08-21 at 4 18 28 PM" src="https://github.com/user-attachments/assets/b7594693-1fd6-4206-b518-79f9a6a7c401">

Reply:
@jonwaldstein Oh ok, I had misunderstood it before. Resolve in 6cfe563590cc2ecf0c54542b5b113d6a5393ebef